open-metadata-implementation/common-services/generic-handlers/src/main/java/org/odpi/openmetadata/commonservices/generichandlers/SchemaAttributeHandler.java

```diff
@@ -1865,6 +1865,142 @@ public void   updateSchemaAttribute(String               userId,
+                                                                          null,
+                                                                          null,
+                                                                          forLineage,
+                                                                          forDuplicateProcessing,
+                                                                          effectiveTime,
+                                                                          methodName);
+
+        if (schemaAttributeEntity != null)
+        {
+            InstanceProperties instanceProperties = schemaAttributeBuilder.getInstanceProperties(methodName);
+
+            this.updateBeanInRepository(userId,
+                    externalSourceGUID,
+                    externalSourceName,
+                    schemaAttributeGUID,
+                    schemaAttributeGUIDParameterName,
+                    OpenMetadataAPIMapper.SCHEMA_ATTRIBUTE_TYPE_GUID,
+                    OpenMetadataAPIMapper.SCHEMA_ATTRIBUTE_TYPE_NAME,
+                    forLineage,
+                    forDuplicateProcessing,
+                    supportedZones,
+                    instanceProperties,
+                    true,
+                    effectiveTime,
+                    methodName);
+
+            SchemaTypeBuilder schemaTypeBuilder = schemaAttributeBuilder.getSchemaTypeBuilder();
+            if (schemaTypeBuilder != null)
+            {
+                // todo this logic assumes the schema type is stored as a classification
+                setClassificationInRepository(userId,
+                                              externalSourceGUID,
+                                                      externalSourceName,
+                                                      schemaAttributeEntity,
+                                                      schemaAttributeGUIDParameterName,
+                                                      attributeTypeName,
+                                                      OpenMetadataAPIMapper.TYPE_EMBEDDED_ATTRIBUTE_CLASSIFICATION_TYPE_GUID,
+                                                      OpenMetadataAPIMapper.TYPE_EMBEDDED_ATTRIBUTE_CLASSIFICATION_TYPE_NAME,
+                                                      schemaTypeBuilder.getTypeEmbeddedInstanceProperties(methodName),
+                                                      isMergeUpdate,
+                                                      forLineage,
+                                                      forDuplicateProcessing,
+                                                      supportedZones,
+                                                      effectiveTime,
+                                                      methodName);
+            }
+            /*
+             * The formula is set if the schema attribute is derived
+             */
+            if (formula != null && schemaTypeBuilder != null)
```

Comment:
Move action for formula inside previous if statement so only test for null schemaTypeBuilder once.

Reply:
I am not sure about the logic around the formula, it seems to be setting content in the schemaAttributeBuilder but that variable is never used.

Reply:
I have redone the logic. I am putting back to draft - so I can do more testing